File: Js-Matrices-Multi/RegistroPersona.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let matriz = [];

function mostrarMenu() {
  console.log("\n--- Menú ---");
  console.log("1. Ingresar nueva persona");
  console.log("2. Mostrar todos los datos");
  console.log("3. Filtrar por DNI");
  console.log("4. Salir");
}

function ingresarPersona() {
  rl.question('Ingresa el nombre: ', (nombre) => {
    rl.question('Ingresa el apellido: ', (apellido) => {
      rl.question('Ingresa el DNI: ', (dni) => {
        rl.question('Ingresa los teléfonos separados por comas: ', (telefonos) => {
          rl.question('Ingresa los nombres de los hijos separados por comas: ', (hijos) => {
            const persona = [nombre, apellido, dni, telefonos.split(','), hijos.split(',')];
            matriz.push(persona);
            console.log("Persona ingresada exitosamente.");
            mostrarMenuYEsperarOpcion();
          });
        });
      });
    });
  });
}

function mostrarDatos() {
  if (matriz.length === 0) {
    console.log("No hay datos para mostrar.");
    mostrarMenuYEsperarOpcion();
    return;
  }

  console.log("\nDatos ingresados:");
  matriz.forEach(persona => {
    const [nombre, apellido, dni, telefonos, hijos] = persona;
    console.log(`${nombre} ${apellido}, DNI: ${dni}, Teléfonos: ${telefonos.length} teléfono(s), Hijos: ${hijos.length}`);
  });

  console.log("\nMatriz de datos:");
  console.log(matriz);
  mostrarMenuYEsperarOpcion();
}

function filtrarPorDNI() {
  rl.question('Ingresa el DNI para filtrar: ', (dni) => {
    const persona = matriz.find(p => p[2] === dni);
    if (persona) {
      const [nombre, apellido, _, telefonos, hijos] = persona;
      console.log(`\nDatos de ${nombre} ${apellido}:`);
      console.log(`DNI: ${dni}, Teléfonos: ${telefonos.length} teléfono(s), Hijos: ${hijos.length}`);
    } else {
      console.log("No se encontró una persona con ese DNI.");
    }
    mostrarMenuYEsperarOpcion();
  });
}

function mostrarMenuYEsperarOpcion() {
  mostrarMenu();
  rl.question('Elige una opción: ', (opcion) => {
    switch (opcion) {
      case '1':
        ingresarPersona();
        break;
      case '2':
        mostrarDatos();
        break;
      case '3':
        filtrarPorDNI();
        break;
      case '4':
        console.log("Saliendo del programa.");
        rl.close();
        break;
      default:
        console.log("Opción no válida. Por favor, elige una opción válida.");
        mostrarMenuYEsperarOpcion();
    }
  });
}

mostrarMenuYEsperarOpcion();

File: Python-Matrices-Multi/RegistroPersona.py
def mostrar_menu():
    print("-- Bienvenido al Menu----")
    print("1. Ingresar nueva persona")
    print("2. Mostrar todos los datos")
    print("3. Filtrar por DNI")
    print("4. Salir")
    
def Ingresar_persona(matriz):
    nombre = input("Ingrese el nombre: ")
    apellido = input("Ingrese el apellido: ")
    dni = input("Ingrese el DNI: ")
    telefono = input("Ingrese el telefono: ").split(",")
    hijos = input("Ingrese los nombre de los hijos: ").split(",")
    
    persona = [nombre, apellido, dni, telefono, hijos]
    matriz.append(persona)
    print("Persona ingresada exitosamente.")
    
def mostrar_persona(matriz):
    if not matriz:
        print("No hay personas Registrdas")
        return
    print("--- Personas Registradas ---")
    for persona in matriz:
        nombre,apellido,dni,telefono,hijos = persona
        print(f"Nombre: {nombre}, Apellido: {apellido}, DNI: {dni}, Telefono: {len(telefono)}, Hijos: {len(hijos)}")
    
    print("\nMatriz de datos:")
    for persona in matriz:
        print(persona)
        
def filtrar_persona(matriz):
    dni = input("Ingrese el DNI a buscar: ")
    for persona in matriz:
        if persona[2] == dni:
           nombre, apellido, _, telefonos, hijos = persona
           print(f"\nDATOS DE {nombre} {apellido}")
           print(f"DNI: {dni}, Teléfonos: {len(telefonos)} teléfono(s), Hijos: {len(hijos)}")
           return
    print("No se encontró una persona con ese DNI.")
    
def main():
    matriz = []
    while True:
        mostrar_menu()
        opcion = input("Seleccione una opció: ")
        if opcion == "1":
            Ingresar_persona(matriz)
        elif opcion == "2":
            mostrar_persona(matriz)
        elif opcion == "3":
            filtrar_persona(matriz)
        elif opcion == "4":
            print("Saliendo del Programa....")
            break
        else:
            print("Opcion no válida, por favor seleccione una opció válida.")
            
if __name__ == "__main__":
    main()